x-pack/plugins/endpoint/server/routes/alerts.ts

```diff
@@ -5,27 +5,92 @@
  */
 
 import { IRouter } from 'kibana/server';
+import { RequestHandler } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { schema } from '@kbn/config-schema';
 
-import json from './sampledata.json';
+import { kibanaRequestToAlertListQuery } from '../services/endpoint/alert_query_builders';
+import { AlertData, AlertResultList } from '../../common/types';
+import { EndpointAppContext } from '../types';
+
+const ALERTS_ROUTE = '/api/endpoint/alerts';
+
+export function registerAlertRoutes(router: IRouter, endpointAppContext: EndpointAppContext) {
+  const alertsHandler: RequestHandler<unknown, unknown, unknown> = async (ctx, req, res) => {
+    try {
+      const queryParams = await kibanaRequestToAlertListQuery(req, endpointAppContext);
+      const response = (await ctx.core.elasticsearch.dataClient.callAsCurrentUser(
+        'search',
+        queryParams
+      )) as SearchResponse<AlertData>;
+      return res.ok({ body: mapToAlertResultList(endpointAppContext, queryParams, response) });
+    } catch (err) {
+      return res.internalError({ body: err });
+    }
+  };
 
-export function registerAlertRoutes(router: IRouter) {
   router.get(
     {
-      path: '/api/endpoint/alerts',
-      validate: false,
+      path: ALERTS_ROUTE,
+      validate: {
+        query: schema.object({
+          page_size: schema.number({ defaultValue: 10, min: 1, max: 10000 }),
```

Comment:
Can you write a negative test where page_size is negative or >10000?

Reply:
Yeah, negative tests are TBD. I left myself a comment to do it.